File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/java/matching.java
/**
 * Java Implementation: Bipartite Matching (Hopcroft-Karp)
 * Invariant: O(E sqrt(V)) maximum cardinality matching
 */
package seed.dsa.core;

import java.util.*;

public class BipartiteMatchingHopcroftKarp<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}

File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/csharp/matching.cs
/**
 * C# Implementation: Bipartite Matching (Hopcroft-Karp)
 * Invariant: O(E sqrt(V)) maximum cardinality matching
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class BipartiteMatchingHopcroftKarp<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}
